distributed/utils.py

```diff
@@ -696,3 +700,46 @@ def open_port(host=''):
     port = s.getsockname()[1]
     s.close()
     return port
+
+
+def import_file(path):
+    """ Loads modules for a file (.py, .pyc, .zip, .egg) """
+    directory, filename = os.path.split(path)
+    name, ext = os.path.splitext(filename)
+    names_to_import = []
+    tmp_python_path = None
+
+    if ext in ('.py', '.pyc'):
+        if directory not in sys.path:
+            tmp_python_path = directory
+        names_to_import.append(name)
+        # Ensures that no pyc file will be reused
+        cache_file = cache_from_source(path)
+        if os.path.exists(cache_file):
+            os.remove(cache_file)
+    if ext in ('.egg', '.zip'):
+        if path not in sys.path:
+            sys.path.insert(0, path)
```

Comment:
Ok, I understand this case is different (we're adding a given package to the path, not a whole directory). My bad.